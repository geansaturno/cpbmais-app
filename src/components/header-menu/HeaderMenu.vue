<template>
  <ul class="menu menu__wrapper ">
    <li
      v-for="item in menu"
      :key="item.label"
      class="menu__item"
    >
      <router-link :to="item.link" :is="item.link ? 'router-link': 'span'" class="">
        {{item.label}}
        <ul v-if="item.internal">
          <li
            v-for="internal in item.internal"
            :key="internal.name"
          >
            <router-link :to="internal.link" :is="internal.link ? 'router-link': 'span'" class="menu__internal">
              {{internal.label}}
            </router-link>
          </li>
        </ul>
      </router-link>
    </li>
  </ul>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import { MenuItem } from '@/custom-types'

@Component({
  name: 'HeaderMenu'
})
export default class HelloWorld extends Vue {
  @Prop({ required: true })
  menu!: MenuItem;
}
</script>

<style lang="scss">
@import 'src/assets/scss/variables.scss';

.menu {
  background-color: $g3;
  max-width: 350px;
  width: 100vw;

  &__item {
    color: $g1;
    font-size: 21px;
    padding: 19px 0;
    margin: 0 22px;

    &:not(:last-child) {
      border-bottom: 1px solid $g2;
    }
  }

  &__wrapper {
    list-style: none;
    padding-left: 0;

    &--internal {
      padding-left: 30px;
    }
  }
}

</style>
