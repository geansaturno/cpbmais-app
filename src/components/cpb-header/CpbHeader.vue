<template>
  <nav class="header">
      <button class="header__menuButton" @click="toogleMenu">
        <menu-icon class="header__menuIcon" v-if="!showMenu"/>
        <close-icon class="header__menuIcon" v-if="showMenu"/>
      </button>
      <header-menu v-show="showMenu" :menu="menu" class="header__menu"/>
      <img src="../../assets/logoCPBmais.png" alt="" class="header__logo">
  </nav>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator'
import HeaderMenu from '../header-menu/HeaderMenu.vue'
import MenuIcon from '../../assets/svg/menu.svg'
import CloseIcon from '../../assets/svg/close.svg'
import { mapState } from 'vuex'

@Component({
  components: { HeaderMenu, MenuIcon, CloseIcon },
  name: 'CpbHeader',
  computed: {
    ...mapState(['menu'])
  }
})
export default class CpbHeader extends Vue {
  private showMenu = false

  toogleMenu (): void {
    this.showMenu = !this.showMenu
  }
}
</script>

<style lang="scss">
@import 'src/assets/scss/variables.scss';

.header {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: $color-primary;
  position: relative;
  height: 72px;

  &__menu {
    position: absolute;
    top: 72px;
    left: 0;
    height: calc(100vh - 72px);
  }

  &__menuButton {
    background-color: initial;
    border: none;
    padding: 15px;
    position: absolute;
    left: 0;
  }

  &__menuIcon {
    width: 30px;
    color: $g1;
  }

  &__logo {
    margin-left: 35px;
  }
}
</style>
