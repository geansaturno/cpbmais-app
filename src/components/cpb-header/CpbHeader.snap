// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CpbHeader.vue Deve renderizar o heder 1`] = `
<nav class="header"><button class="header__menuButton">
    <!---->
  </button>
  <header-menu-stub menu="[object Object],[object Object]" style="display: none;"></header-menu-stub> <img src="../../assets/logoCPBmais.png" alt="" class="header__logo">
</nav>
`;
